### PR TITLE
Use SharedPreferencesForWebProcess in Network process

### PR DESCRIPTION
**Pull request by rniwa**
#### e0533e5d92ac581fc2f18039c3b2fdbd0fc9db53
<pre>
Use SharedPreferencesForWebProcess in Network process
<a href="https://bugs.webkit.org/show_bug.cgi?id=277106">https://bugs.webkit.org/show_bug.cgi?id=277106</a>

Reviewed by Chris Dumez.

Use SharedPreferencesForWebProcess added in <a href="https://commits.webkit.org/281360@main">https://commits.webkit.org/281360@main</a> for Network
process in place of NetworkProcessPreferencesForWebProcess.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp:
(WebKit::m_sharedPreferencesForWebProcess):
(WebKit::m_allowTestOnlyIPC): Deleted.
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.h:
(WebKit::NetworkConnectionToWebProcess::sharedPreferencesForWebProcess const):
(WebKit::NetworkConnectionToWebProcess::updateSharedPreferencesForWebProcess):
(WebKit::NetworkConnectionToWebProcess::isWebTransportEnabled const):
(WebKit::NetworkConnectionToWebProcess::usesSingleWebProcess const):
(WebKit::NetworkConnectionToWebProcess::allowTestOnlyIPC const):
* Source/WebKit/NetworkProcess/NetworkProcess.cpp:
(WebKit::NetworkProcess::sharedPreferencesForWebProcessDidChange):
* Source/WebKit/NetworkProcess/NetworkProcess.h:
* Source/WebKit/NetworkProcess/NetworkProcess.messages.in:
* Source/WebKit/Shared/NetworkProcessConnectionParameters.h:
* Source/WebKit/Shared/NetworkProcessConnectionParameters.serialization.in:
* Source/WebKit/Shared/NetworkProcessPreferencesForWebProcess.h: Removed.
* Source/WebKit/Shared/NetworkProcessPreferencesForWebProcess.serialization.in: Removed.
* Source/WebKit/UIProcess/API/APIPageConfiguration.cpp:
(API::PageConfiguration::preferencesForNetworkProcess const): Deleted.
* Source/WebKit/UIProcess/API/APIPageConfiguration.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _initializeWithConfiguration:]):
(-[WKWebView _setupPageConfiguration:withPool:]):
(-[WKWebView _setupPageConfiguration:]): Deleted.
* Source/WebKit/UIProcess/Network/NetworkProcessProxy.cpp:
(WebKit::NetworkProcessProxy::getNetworkProcessConnection):
(WebKit::NetworkProcessProxy::sharedPreferencesForWebProcessDidChange):
* Source/WebKit/UIProcess/Network/NetworkProcessProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::hasSameGPUAndNetworkProcessPreferencesAs const):
(WebKit::WebPageProxy::preferencesDidChange):
(WebKit::WebPageProxy::preferencesForNetworkProcess const): Deleted.
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebProcessPool.cpp:
(WebKit::WebProcessPool::establishRemoteWorkerContextConnectionToNetworkProcess):
(WebKit::WebProcessPool::tryTakePrewarmedProcess):
(WebKit::WebProcessPool::processForRegistrableDomain):
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::initializePreferencesForGPUAndNetworkProcesses):
(WebKit::WebProcessProxy::hasSameGPUAndNetworkProcessPreferencesAs const):
(WebKit::WebProcessProxy::didSyncSharedPreferencesForWebProcessWithNetworkProcess):
(WebKit::WebProcessProxy::didSyncSharedPreferencesForWebProcessWithGPUProcess):
(WebKit::WebProcessProxy::waitForSharedPreferencesForWebProcessToSync):
(WebKit::WebProcessProxy::initializePreferencesForNetworkProcess): Deleted.
(WebKit::WebProcessProxy::didSyncSharedPreferencesForWebProcess): Deleted.
* Source/WebKit/UIProcess/WebProcessProxy.h:
(WebKit::WebProcessProxy::preferencesForNetworkProcess const): Deleted.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/281415@main">https://commits.webkit.org/281415@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/ba4c5a1eb6ed0d6193249f12265d65ca029694f2

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/59731 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/39077 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/12259 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/63647 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/10255 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/46729 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/10415 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/48443 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/7169 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/61761 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/36463 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/51715 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/29282 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/33169 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/8959 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/9178 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/52825 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/55103 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/9238 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/65378 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/58976 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/3659 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/9109 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/55783 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/3670 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/51700 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/55918 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/3039 "Passed tests") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/80734 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8957 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/34890 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/14012 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/35973 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/37059 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/35718 "Built successfully") | | |

### CODE REVIEW

File: Source/WebKit/UIProcess/WebPageProxy.cpp [hunk 1]
```diff
@@ -1178,12 +1178,8 @@ void WebPageProxy::handleSynchronousMessage(IPC::Connection& connection, const S
 
 bool WebPageProxy::hasSameGPUAndNetworkProcessPreferencesAs(const API::PageConfiguration& configuration) const
```
Comment (cdumez) [inline Source/WebKit/UIProcess/WebPageProxy.cpp:1179]: We should check (in a follow-up) if we still need this function. Previously, those shared preferences could not be updated so we could not add to a WebProcess a WebPage with different preferences than the previous pages in this process. However, your new model supports updating those preferences so we may be able to reuse WebProcesses more aggressively, like we used to before my changes.

File: Source/WebKit/UIProcess/WebProcessProxy.cpp [hunk 2]
```diff
@@ -491,25 +486,8 @@ void WebProcessProxy::initializePreferencesForGPUAndNetworkProcesses(const WebPa
...
-void WebProcessProxy::initializePreferencesForNetworkProcess(const WebPreferencesStore& preferences)
-{
-    ASSERT(!m_preferencesForNetworkProcess);
-    m_preferencesForNetworkProcess = NetworkProcessPreferencesForWebProcess {
-        preferences.getBoolValueForKey(WebPreferencesKey::webTransportEnabledKey()),
-        processPool().usesSingleWebProcess(),
-    };
-}
-
 bool WebProcessProxy::hasSameGPUAndNetworkProcessPreferencesAs(const API::PageConfiguration& pageConfiguration) const
```
Comment (cdumez) [inline Source/WebKit/UIProcess/WebProcessProxy.cpp:489]: Ditto, we should check if this is really still needed with your new approach.

File: Source/WebKit/UIProcess/Network/NetworkProcessProxy.cpp [hunk 3]
```diff
@@ -308,14 +308,13 @@ void NetworkProcessProxy::getNetworkProcessConnection(WebProcessProxy& webProces
     RELEASE_LOG(ProcessSuspension, "%p - NetworkProcessProxy is taking a background assertion because a web process is requesting a connection", this);
     startResponsivenessTimer(UseLazyStop::No);
     NetworkProcessConnectionParameters parameters;
-    parameters.allowTestOnlyIPC = webProcessProxy.allowTestOnlyIPC();
 #if ENABLE(IPC_TESTING_API)
     parameters.ignoreInvalidMessageForTesting = webProcessProxy.ignoreInvalidMessageForTesting();
 #endif
     // The WebProcess shouldn't request a connection to the network process before we've had a chance to
```
Comment (cdumez) [inline Source/WebKit/UIProcess/Network/NetworkProcessProxy.cpp]: I think this comment is outdated now? At least we dropped the assertion below that enforced it.

**Review by cdumez — APPROVED** [review]

File: Source/WebKit/UIProcess/Network/NetworkProcessProxy.cpp (hunk 3 above)
Comment (rniwa) [reply Source/WebKit/UIProcess/Network/NetworkProcessProxy.cpp]: Will remove.

File: Source/WebKit/UIProcess/WebPageProxy.cpp (hunk 1 above)
Comment (rniwa) [reply Source/WebKit/UIProcess/WebPageProxy.cpp:1179]: Good point. We should reconsider that.

**Pull request merged**